app/models/user.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  attr_accessor :remember_token
+
+  before_create :create_token
+  has_secure_password
+  has_many :posts
+
+  validates :name, presence: true, length: { in: 4..30 }
+  validates :email, presence: true, format: /\w+@\w+\.{1}[a-zA-Z]{2,}/
```

Comment:
Although not a requirement for this project, it is advisable to guaranteeing uniqueness on fields that identify an user. More so if the field is user for authentication.

[`OPTIONAL`]